go-controller/pkg/ovn/policy.go

```diff
@@ -168,18 +168,27 @@ func (oc *Controller) addIPBlockACLDeny(np *namespacePolicy,
 	return
 }
 
-func (oc *Controller) addACLDenyPortGroup(portGroupUUID, portGroupName,
-	priority string, policyType knet.PolicyType) error {
-	var match, direction string
-	direction = toLport
+func (oc *Controller) addACLPortGroup(portGroupUUID, portGroupName,
+	direction, priority, extraMatch, action string,
+	policyType knet.PolicyType) error {
+	var match string
+
+	if extraMatch != "" {
+		extraMatch = fmt.Sprintf(" && %s", extraMatch)
+	}
+
 	if policyType == knet.PolicyTypeIngress {
-		match = fmt.Sprintf("match=\"outport == @%s\"", portGroupName)
+		match = fmt.Sprintf("match=\"outport == @%s%s\"", portGroupName,
+			extraMatch)
 	} else {
-		match = fmt.Sprintf("match=\"inport == @%s\"", portGroupName)
+		match = fmt.Sprintf("match=\"inport == @%s%s\"", portGroupName,
+			extraMatch)
 	}
 
```

Comment:
@dceara instead of changing the if block you could:

```
if policyType == knet.PolicyTypeIngress {
   ...
} else {
   ...
}

if extraMatch != "" {
    match = match + " && " + extraMatch
}
```

Reply:
@dcbw I had to do it in a bit of a different way because ovn expects the string `match="<match-expr>"`. So in the latest version I first build the `match-expr` and then put it between quotes.